packages/find-and-replace/FindAndReplaceManager.js

```diff
@@ -209,20 +219,24 @@ class FindAndReplaceManager {
   }
 
   _propagateUpdate(renderSelection) {
+    // HACK: we make commandStates dirty in order to trigger re-evaluation
+    this._updateMarkers()
+    this.editorSession._setDirty('commandStates')
+    if (!renderSelection) {
+      this.editorSession.skipNextSelectionRerender()
```

Comment:
Just don't think skipping next is enough. E.g. think if a reflow happens because of collaboration stuff. Then you would loos the cursor too.
I think the only valid implementation is to have an explicit app state. Even if that means that the app state needs to be managed carefully.